pkg/common/kafka/testing/mocks.go

```diff
@@ -18,47 +18,57 @@ package testing
 
 import (
 	"context"
-	"testing"
+	"fmt"
 
 	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/mock"
 )
 
 //
-// Mock ConsumerGroup Implementation
+// Mock Sarama ConsumerGroup Implementation
 //
 
-// Verify The Mock ConsumerGroup Implements The Interface
 var _ sarama.ConsumerGroup = &MockConsumerGroup{}
 
-// Define The Mock ConsumerGroup
 type MockConsumerGroup struct {
-	t           *testing.T
-	errorChan   chan error
+	mock.Mock
+	ErrorChan   chan error
 	consumeChan chan struct{}
 	Closed      bool
+	CloseErr    bool
 }
 
-// Mock ConsumerGroup Constructor
-func NewMockConsumerGroup(t *testing.T) *MockConsumerGroup {
-	return &MockConsumerGroup{
-		t:           t,
-		errorChan:   make(chan error),
+func NewStubbedMockConsumerGroup() *MockConsumerGroup {
+	mockGroup := &MockConsumerGroup{
+		ErrorChan:   make(chan error),
 		consumeChan: make(chan struct{}),
 		Closed:      false,
 	}
+
+	// Stub out the mock functions since this mock returns "real" data
+	mockGroup.On("Consume", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	mockGroup.On("Errors").Return(nil)
+	mockGroup.On("Close").Return(nil)
+	return mockGroup
 }
 
 func (m *MockConsumerGroup) Consume(ctx context.Context, topics []string, handler sarama.ConsumerGroupHandler) error {
+	_ = m.Called(ctx, topics, handler)
 	<-m.consumeChan                      // Block To Simulate Real Execution
 	return sarama.ErrClosedConsumerGroup // Return ConsumerGroup Closed "Error" For Clean Shutdown
 }
```

Comment:
seems odd to hard code a response when using the Testify mocks instead of... 
```
args := m.Called(ctx, topics, handler)
<-m.consumeChan
return args.Error(0)
```

Reply:
These are all because I back-filled the testify stuff into hard-coded mocks; I'll clean them up to be testify-style-only